contact_manager.py

```diff
@@ -0,0 +1,218 @@
+                    details = c.get_details(phone_no)
+                    print('Name: {name}, Phone_no: {phone_no}, Email: {email}, Street: {street}, City: {city}, '
+                          'State: {state}, Pin_code: {pin_code}'.format(name=details.name,
+                                                                        phone_no=details.phone_no,
+                                                                        email=details.email,
+                                                                        street=details.address.street,
+                                                                        city=details.address.city,
+                                                                        state=details.address.state,
+                                                                        pin_code=details.address.pin_code))
+                else:
+                    print('There is no contact with this phone number!')
+            elif ip == '5':
+                phone_no = input('Phone_no:')
+                if len(phone_no) == 10 and phone_no[0] != '0' and phone_no.isnumeric():
+                    provider_name = c.get_provider(phone_no)
+                    print(provider_name)
+                else:
+                    print('Phone number is not valid!')
+            elif ip == '6':
+                provider_name = input('Provider name:')
+                if provider_name in c.provider.keys():
+                    records = c.get_records_of_provider(provider_name)
+                    for item in records:
+                        print('Name: {name}, Phone_no: {phone_no}, Email: {email}, Street: {street}, City: {city}, '
+                              'State: {state}, Pin_code: {pin_code}'.format(name=item.name,
+                                                                            phone_no=item.phone_no,
+                                                                            email=item.email,
+                                                                            street=item.address.street,
+                                                                            city=item.address.city,
+                                                                            state=item.address.state,
+                                                                            pin_code=item.address.pin_code))
+
+            elif ip == '7':
+                fields = ['name', 'phone_no', 'email', 'street', 'city', 'state', 'pin_code']
+                st = input('Enter a string:')
+                field = input('Enter a field:')
+                if field in fields:
+                    li = c.get_records_containing_string(st, field)
+                    for item in li:
+                        print('Name: {name}, Phone_no: {phone_no}, Email: {email}, Street: {street}, City: {city}, '
+                              'State: {state}, Pin_code: {pin_code}'.format(name=item.name,
```

Comment:
expose a method call get_details_string / get_formatted_details and call it here!!